assets/js/blocks/products-by-attribute/block.tsx

```diff
@@ -208,24 +206,46 @@ class ProductsByAttributeBlock extends Component {
 				) }
 			</>
 		);
-	}
+	};
+
+	return render();
+};
+
+interface Attributes {
+	align?: string;
+	attributes: Array< string >;
+	attrOperator: string;
+	columns: number;
+	editMode: boolean;
+	contentVisibility: {
+		image: boolean;
+		title: boolean;
+		price: boolean;
+		rating: boolean;
+		button: boolean;
+	};
+	orderby: string;
+	rows: number;
+	alignButtons: boolean;
+	isPreview: boolean;
+	stockStatus: Array< string >;
 }
 
-ProductsByAttributeBlock.propTypes = {
+interface Props {
```

Comment:
I'd love if we extracted this as a shared type as we've talked about and potentially pushed it upstream to Gutenberg.

We could type attributes with a generic type.

The `debouncedSpeak` comes from `withSpokenMessages` HOC so it's not shared.

I've tried to follow what is provided by default and did a dump of props for this block. A lot of these are shortcuts to actions on the blocks data store or flags based on the editor state ie. `isSelected`.  If we had these typed I believe we'd feel more comfortable using these :) 

I think for the minimum we should provide `name`, `clientId`, `attributes`, `setAttributes`, `isSelected` - though having other typed would be nice as well. We can split it as a separate PR maybe as it might require a bit of research on the types and which (if any) are already typed in Gutenberg or in DT

```
{ 
   attributes: {alt: '', contentAlign: 'center', dimRatio: 50, editMode: false, focalPoint: false, …}
   clientId: "1d574829-a8a2-49c9-a887-6f5735d278a6"
   context: {}
   insertBlocksAfter: ƒ ()
   isSelected: false
   isSelectionEnabled: true
   mergeBlocks: ƒ ()
   name: "woocommerce/featured-product"
   onRemove: () => removeBlock(clientId)
   onReplace: ƒ ()
   setAttributes: ƒ ()
   toggleSelection: ƒ ()
```



One thing regarding the Attributes type name to keep in mind is that it is distinct from BlockAttributes that we pass to a block when registering which holds attribute configuration. This one is provided by `@types/wordpress__blocks`